[13.0][IMP] l10n_es_aeat_mod303: New 2021/07 format

pedrobaeza opened the pull request:
Forward-port + adaptation of #1777 

@Tecnativa TT31344

angelmoya committed:
[IMP] l10n_es_aeat_mod303: Add support for new BOE 2021/07 format
    * New tax mapping, field 84 now split on 84 and 125
    * New export config, now in csv
    * New marca_sepa field
    * Update field 80

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod303: Finish 2021-07 format

* Transform tax map to CSV
* Incorrect BOE export definition
* Simplify certain expressions
* Move CSV to subdirectory to classify them by format change batch
* End date for previous BOE format
* Fix marca_sepa computation
* Restore back field 85 for older reports

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: v13 adaptation to 2021-07 format

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Adopt it as maintainer

acysos commented:
Hola,

Mismo error en el archivo E030070 que en la 12.0, a vosotros no os da?

Saludos

pedrobaeza committed:
[FIX] l10n_es_aeat: Non stored + archive ribbon

- Always assign value to non stored computed field
- Put archived ribbon ala v13 style
- Adopt it as maintainer

pedrobaeza commented:
@acysos el validador que apuntaste el otro día no sirve. No sé si lo estás probando con eso. Si lo haces de manera regular, indica por favor el error. Yo ahora mismo he podido hacer una declaración, pero con otra estoy teniendo el error `Contenido incorrecto en 'Contenido obligatorio 0'`, y estoy tratando de ver cuál es el problema. En v12 pude realizar dos declaraciones sin problemas.

acysos commented:
@pedrobaeza Lo se, estoy usando el nuevo y da ese error. He quitado ese valor tal y como indica AEAT y funciona. El problema es la casilla 61 ya no se utiliza en el nuevo formato. Si las pruebas que has hecho no tienen valor la casila 61 no es problema, pero si la tiene no te deja.

He quitado los impuestos a la casilla 61 para que sea 0 y como bien indicas te deja. Este campo a partir de julio es vacio. Viendo el mapeo de impuestos veo que se le asigna valor a esta casilla Extracomunitarios servicios y ISP que no es correcto. 

Tengo mail de agencia tributaria que indica que no se debe usar esa casilla ya.

acysos commented:
@pedrobaeza Veo que las casillas 120 y 122 no tienen impuestos, que son justamente los de la casilla 61. Preparo mejor un PR? Que veo que son varios cambios.

acysos commented:
Tengo la declaración especial. Un tercer error:

1 | E030393 | El campo 'Devolución - Marca SEPA' no puede tener contenido si la declaración no es a devolver
-- | -- | --

Los dos anteriores los tengo resueltos voy a por esta a ver que pasa y mando el PR.

acysos committed:
[FIX] l10n_es_aeat_mod303: Incorrect taxes + marca_sepa

Remove taxes from field 61, and add them to both fields 120 y 122.
That's because since 2021/07, these fields contains such values, and as
the tax mapping is only used since this date, it's safe to do it.

Something similar applies to marca_sepa, as even if doing a previous
report, this field is not used at all, so we can remove the initial
condition and put the value only when performing return.

acysos commented:
He mandado el PR a la 12.0, para que puedas incorporarlo aquí o prefieres que te lo mande también aquí? https://github.com/OCA/l10n-spain/pull/1806

pedrobaeza commented:
Gracias por la información, Ignacio, resulta muy útil. Reviso y te cuento.

pedrobaeza commented:
Probado en el conflictivo y funcionando!

/ocabot merge major

pedrobaeza commented:
Realizados cherry-picks + adaptación y empujado directamente a la rama 14.0, comprobando tests.
